### PR TITLE
Smol

### PR DESCRIPTION
**Pull request by radex**
- Completely got rid of `rambdax` in WatermelonDB to make its bundle size smaller. I removed most FP helpers that were only used in one or two places and rewrote it as simple vanilla JS for ultimate performance ;) The few rambdax helpers that I did absolutely need, I rewrote myself and added tests. 
- Took a bit better care of what's imported where to keep the critical path for loading IDB on web as short as possible (since loading IDB is concurrent and takes the longest of the whole launch)

### CODE REVIEW

**Review by kokusGr — APPROVED** [review]
LGTM 👍

**Comment by kokusGr** [comment]
@radex Not sure if I should merge it since this is a dependant branch so I assign this to you.

File: src/adapters/lokijs/worker/lokiExtensions.js [hunk 1]
```diff
@@ -56,8 +56,8 @@ async function getLokiAdapter(options: LokiAdapterOptions): mixed {
     return adapter
   } else if (await isIDBAvailable(onQuotaExceededError)) {
     if (useIncrementalIndexedDB) {
-      const IncrementalIDBAdapter = options._concurrentIdb ?
-        require('lokijs-concurrent-idb/src/incremental-indexeddb-adapter') :
+      const IncrementalIDBAdapter = options._betaLoki ?
+        require('lokijs/src/incremental-indexeddb-adapter') :
```
[the remark this answers is not in the document]
Comment (radex) [reply src/adapters/lokijs/worker/lokiExtensions.js:60]: yep, I figured I'll surely need to test different versions of loki in the future...

**Comment by radex** [comment]
ah right, I forgot - in this case, I'm assigning the other one to you too: https://github.com/Nozbe/WatermelonDB/pull/935 ;)

**Pull request merged**